extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -163,6 +170,64 @@ class CornerstoneViewportService implements IViewportService {
 
     this.viewportsInfo.get(viewportIndex).destroy();
     this.viewportsInfo.delete(viewportIndex);
+    this.viewportsById.delete(viewportId);
+  }
+
+  public getPresentation(viewportIdx: number): Presentation {
+    const viewportInfo = this.viewportsInfo.get(viewportIdx);
+    if (!viewportInfo) return;
+    const {
+      presentationId: id,
+      viewportType,
+    } = viewportInfo.getViewportOptions();
+    if (!id) return;
+
+    const csViewport = this.getCornerstoneViewportByIndex(viewportIdx);
+    if (!csViewport) return;
+
+    if (viewportType === 'stack' || !viewportType) {
+      const pan = csViewport.getPan();
+      const zoom = csViewport.getZoom();
+      const properties = csViewport.getProperties();
+      const initialImageIndex = csViewport.getCurrentImageIdIndex();
+      return {
+        id,
+        viewportType: viewportType === 'volume3d' ? 'volume' : 'stack',
+        pan,
+        zoom,
+        properties,
+        initialImageIndex,
+      };
+    }
+
+    const camera = csViewport.getCamera();
+    return {
+      id,
+      viewportType: 'volume',
+      camera,
+    };
+  }
+
+  public setPresentation(
+    viewportIdx: number,
+    presentation: Presentation
+  ): void {
+    if (!presentation) return;
+    const viewport = this.getCornerstoneViewportByIndex(viewportIdx);
+    if (presentation.viewportType === 'stack') {
+      const { zoom, pan, properties } = presentation;
+      if (zoom && !isNaN(zoom)) {
```

Comment:
why we have a check here, since we stored them and we know that there is Pan, zoom and properties